### ISSUE TITLE
Ability to enable remote state sharing

### ISSUE THREAD

**Issue by jderieg** [AzBuilder/terrakube#457]
### Feature description 💡

Would love the ability to enable states to be shared between workspaces, similar to this functionality on terraform cloud.
![statesharing](https://github.com/AzBuilder/terrakube/assets/63372114/0d7d2711-e8f5-46cf-8764-52a2e14da593)

We have several workspaces that require data from outputs of other state files. In Terrakube, it doesn't look like we have the ability to do that yet.

**Comment by alfespa17** [comment]
Hello @jderieg 

Could you share a little example of how the terraform resource block code should look when sharing the state?

**Comment by jderieg** [comment]
Sure, no problem. This is what the data block looks like when referring to the remote state:

```
data "terraform_remote_state" "my_vpc" {
  backend = "remote"
  config = {
    organization = "exampleorg"
    hostname = "terrakube.example.com"
    workspaces = {
      name = "my-vpc-workspace"
    }
  }
}
```

And then you would call an output from that remote state like this:

`vpc_id = data.terraform_remote_state.my_vpc.outputs.vpc_id`

More info here:
https://developer.hashicorp.com/terraform/language/state/remote-state-data

**Comment by alfespa17** [comment]
Hello @jderieg 

This feature will be available in 2.15.0, it will be released in the following days.

For now it can be tested using the beta version

```
api:
  version: "2.15.0-beta.3"

executor:
  version: "2.15.0-beta.3"
```

You could check PR [#460](https://github.com/AzBuilder/terrakube/pull/460)
